### PR TITLE
refactor: ConfigActivity refactored design 

### PR DESCRIPTION
**Pull request by aquilesTrindade**
# Before 
ConfigActivity bad design
# After: 
ConfigActivity Refactored

### CODE REVIEW

File: app/src/main/res/values-night/themes.xml [hunk 1]
```diff
@@ -36,6 +36,7 @@
         <item name="colorErrorContainer">@color/md_theme_dark_errorContainer</item>
         <item name="colorOnError">@color/md_theme_dark_onError</item>
         <item name="colorOnErrorContainer">@color/md_theme_dark_onErrorContainer</item>
+        <item name="colorBackground">@color/md_theme_dark_background</item>
```
Comment (remaker17) [inline app/src/main/res/values-night/themes.xml:39]: Why do you need to create custom attribute for background?

File: app/src/main/res/values/attrs.xml [hunk 2]
```diff
@@ -124,4 +124,5 @@
     <attr name="colorVioletContainer" format="color" />
     <attr name="colorOnVioletContainer" format="color" />
     <attr name="harmonizeViolet" format="boolean" />
+    <attr name="colorBackground" format="color" />
```
Comment (remaker17) [inline app/src/main/res/values/attrs.xml:127]: ????

File: app/src/main/res/values/themes.xml [hunk 3]
```diff
@@ -61,6 +61,7 @@
         <item name="colorErrorContainer">@color/md_theme_light_errorContainer</item>
         <item name="colorOnError">@color/md_theme_light_onError</item>
         <item name="colorOnErrorContainer">@color/md_theme_light_onErrorContainer</item>
+        <item name="colorBackground">@color/md_theme_light_background</item>
```
Comment (remaker17) [inline app/src/main/res/values/themes.xml:64]: ????

File: app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java [hunk 4]
```diff
@@ -53,11 +79,25 @@ public class ConfigActivity extends Activity {
     public static final String SETTING_SKIP_MAJOR_CHANGES_REMINDER = "skip-major-changes-reminder";
     public static final String SETTING_BLOCKMANAGER_DIRECTORY_PALETTE_FILE_PATH = "palletteDir";
     public static final String SETTING_BLOCKMANAGER_DIRECTORY_BLOCK_FILE_PATH = "blockDir";
+    public static final String SETTING_APP_THEME = "sketchware-theme";
```
Comment (JavkhlanK) [inline app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java:82]: This commit only mentions "redesign" while it also adds a new setting. That is not OK. The commit is definitely not a refactor btw, refactors don't change the behavior or looks of the app.

File: app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java [hunk 5]
```diff
@@ -53,11 +79,25 @@ public class ConfigActivity extends Activity {
...
+    public int selected = 0;
+    public int DEFAULT_BACKGROUND_COLOR = Color.parseColor("#fafafa");
+        
+    public LinearLayout root;
```
Comment (JavkhlanK) [inline app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java:86]: And why public? How about following OOP principles?

**Comment by ilyassesalama** [comment]
Thanks for the contribution, great work. However, as an addition to what others have said, can you try to fix that toolbar as well? it'd be great if it follows the app new design.

Screenshot 2024-05-02 at 10 45 31 AM


Here's an example of how it may look expanded:



and collapsed:




Moreover, it seems like options' description text is bold, why is that?

File: app/src/main/res/values-night/themes.xml (hunk 1 above)
Comment (aquilesTrindade) [reply app/src/main/res/values-night/themes.xml:39]: Because I needed to get this color in the Config screen

**Comment by aquilesTrindade** [comment]
cool man, then I try again, A and the same commits are because I had 1 letter wrong and the other is because there was 1 line the same in 2 files

File: app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java (hunk 4 above)
Comment (aquilesTrindade) [reply app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java:82]: Bro, as I said on discord, I'm making a theme system right?, and I took this code from the repository where I'm doing this, and I tried to remove the lines related to the theme, this line went unnoticed by me

**Comment by aquilesTrindade** [comment]
good, thanks, I can try to make this design with CollapsingToolbar soon, and I'll find a way without having to create new attributes and colors

**Comment by ilyassesalama** [comment]
Why did you close it?

**Pull request closed without merging**